src/Okta.Idx.Sdk/OktaVerify/OktaVerifyAuthenticationOptions.cs

```diff
@@ -0,0 +1,253 @@
+        /// </summary>
+        /// <param name="methodType">The method type.</param>
+        /// <returns>The authentication response.</returns>
+        protected async Task<AuthenticationResponse> SelectOktaVerifyMethodAsync(string methodType)
+        {
+            IdxRequestPayload idxRequestPayload = new IdxRequestPayload();
+            idxRequestPayload.SetProperty("authenticator", new { id = this.CurrentAuthenticator.Id, methodType = methodType });
+            idxRequestPayload.SetProperty("stateHandle", StateHandle);
+
+            var selectAuthenticatorResponse = await SelectAuthenticatorAuthenticateRemediationOption.ProceedAsync(idxRequestPayload);
+            this.ChallengePollRemediationOption = selectAuthenticatorResponse.FindRemediationOption(RemediationType.ChallengePoll);
+            this.ChallengeAuthenticatorRemediationOption = selectAuthenticatorResponse.FindRemediationOption(RemediationType.ChallengeAuthenticator);
+
+            var authenticationResponse = new AuthenticationResponse
+            {
+                IdxContext = IdxContext,
+                AuthenticationStatus = AuthenticationResponse.AuthenticationStatus,
+                CurrentAuthenticator = IdxResponseHelper.ConvertToAuthenticator(selectAuthenticatorResponse.Authenticators.Value, selectAuthenticatorResponse.CurrentAuthenticator.Value),
+            };
+
+            return authenticationResponse;
+        }
+
+        /// <summary>
+        /// Gets the token response.
+        /// </summary>
+        public ITokenResponse TokenInfo { get; internal set; }
+
+        /// <summary>
+        /// Selects the TOTP authentication method.
+        /// </summary>
+        /// <returns>The authentication response.</returns>
+        public async Task<AuthenticationResponse> SelectTotpMethodAsync()
+        {
+            return await SelectOktaVerifyMethodAsync("totp");
+        }
+
+        /// <summary>
+        /// Selects the push authentication method.
+        /// </summary>
+        /// <returns>The authentication response.</returns>
+        public async Task<AuthenticationResponse> SelectPushMethodAsync()
+        {
+            return await SelectOktaVerifyMethodAsync("push");
+        }
+
+        /// <summary>
+        /// Submits the specified TOTP code for authentication.
+        /// </summary>
+        /// <param name="code">The code.</param>
+        /// <returns>The authentication response.</returns>
+        /// <exception cref="ArgumentException">If the challenge authenticator is not valid.</exception>
+        public async Task<AuthenticationResponse> EnterCodeAsync(string code)
+        {
+            if (ChallengeAuthenticatorRemediationOption.Name != RemediationType.ChallengeAuthenticator)
+            {
+                throw new ArgumentException($"Expected remediation option of type '{RemediationType.ChallengeAuthenticator}', the specified remediation option is of type {ChallengeAuthenticatorRemediationOption.Name}.");
```

Comment:
Let's use existing exceptions. I believe we can use UnexpectedRemediationException here.